modules/upgrade/pages/migrate-3x-4x.adoc

```diff
@@ -1,3 +1,446 @@
+
+
+
+[[bp.sec.mgr.migration.pkg.extern]]
+== Packages on External Storage
+
+
+Some installations may store the package data on external storage (for example, NFS mount on [path]``/var/spacewalk/packages``).
+You do not need to copy this data to the new machine.
+Edit the script located in [path]``/usr/lib/susemanager/bin/mgr-setup`` and remove the respective [command]``rsync`` command (located around line 442).
+
+.Mounting External Storage
+[IMPORTANT]
+====
+Make sure your external storage is mounted on the new machine before starting the system for the first time.
+Analogue handling for [path]``/srv/www/htdocs/pub`` if appropriate.
+
+In general, all needed files and directories, not copied by the migration tool, should be copied to the new server manually.
+====
+
+
+
+[[bp.sec.mgr.migration.trouble]]
+== Troubleshooting
+
+=== Broken {webui} after Migration
+
+It is possible that the {webui} may break during migration.
+This behavior is not a bug, but a browser caching issue.
+The new machine has the same host name and IP address as the old machine.
+This duplication can confuse some Web browsers.
+If you experience this issue reload the page.
+For example, in Firefox pressing the key combination kbd:[Ctrl+F5] should resume normal functionality.
+
+
+
+=== Not Enough Disk Space
+
+In case of trouble check available disk space.
+It is recommended to have /var/spacewalk and /var/lib/pgsql on separate (XFS) file systems.
+Make sure to remove the subvolume entry in /etc/fstab for the subvolume of /var/lib/pqsql when using a separate file system and reboot the server first before continuing.
+
+
+
+===  Corrupted Database Dump
+
+Check the output of the following command (replace [literal]``<SUMA_3.2_MACHINE>`` with the actual host name of your 3.2 source machine):
+
+----
+ssh root@<SUMA_3.2_MACHINE> "su -s /bin/bash - postgres -c exit"
+----
+
+This command must not produce any output.
+Output can lead to a corrupted transfer of the archive with the database dump. Re-visit your bash environment on the 3.2 source machine (for example, the [filename]``.bashrc`` file) and make sure no extra text is printed on the shell start.
+
+
+=== Retrying to Set Up the New Server
+
+To retry setting up the new server, perform the following step on the new server machine:
+
+. remove /root/.MANAGER_SETUP_COMPLETE
+. stop postgresql and remove /var/lib/pgsql/data
+. set the hostname correctly (it now has the host name from the old {productname} server)
+. correct the /etc/hosts file
+. on the new server check /etc/setup_env.sh and see if the correct database name is set:
++
+----
+MANAGER_DB_NAME='susemanager'
+----
+. reboot the server before running [command]``mgr-setup`` again.
+
+
+
+// FIXME: 2019-05-016, ke: replace it with version 4 output
+[[bp.sec.mgr.migration.example]]
+== Example Session
+
```

Comment:
This should be left out as it is from a previous migration (from 2.1 to 3). While most of the stuff will still look similar, there are some details (such as version numbers) that do not match. I don't think this output is very useful anyway.

Reply:
ok, it is not very useful, but it helps if the user can compare it with the output he see on his screen. Only if he see the complete msg at the end, he will know that everything(?) was fine (I'll add such a note, then we'll not need the literal output):

```  ============================================================================
  Migration complete.
  Please shut down the old SUSE Manager server now.
  Reboot the new server and make sure it uses the same IP address and hostname
  as the old SUSE Manager server!

  IMPORTANT: Make sure, if applicable, that your external storage is mounted
  in the new server as well as the ISO images needed for distributions before
  rebooting the new server!
  ============================================================================```